docs/nddata/index.rst

```diff
@@ -7,126 +7,137 @@ N-dimensional datasets (`astropy.nddata`)
+data, a ``wcs`` (world coordinate system) and ``uncertainty`` of the data and
+additional ``meta`` attributes:
 
-     >>> ndd_masked = NDData(ndd, mask = ndd.data > 0.9)
+    >>> data = np.array([1,2,3,4])
+    >>> mask = data > 2
+    >>> unit = 'erg / s'
+    >>> from astropy.nddata import StdDevUncertainty
+    >>> uncertainty = StdDevUncertainty(np.sqrt(data)) # representing standard deviation
+    >>> meta = {'object': 'fictional data.'}
+    >>> from astropy.coordinates import SkyCoord
+    >>> wcs = SkyCoord('00h42m44.3s', '+41d16m09s')
+    >>> ndd = NDData(data, mask=mask, unit=unit, uncertainty=uncertainty,
+    ...              meta=meta, wcs=wcs)
+    >>> ndd
+    NDData([1, 2, 3, 4])
 
-A mask value of `True` indicates a value that should be ignored, while a mask
-value of `False` indicates a valid value.
+The representation only displays the ``data``; the other attributes need to be
+accessed directly, for example ``ndd.mask`` to access the mask.
 
 
-Similar attributes are available to store:
+NDDataRef
+---------
 
-+ generic meta-data, in ``meta``,
-+ a unit for the data values, in ``unit`` and
-+ an uncertainty for the data values, in ``uncertainty``. Note that the
-  ``uncertainty`` must have a attribute called ``uncertainty_type`` otherwise
-  it will be converted to an `~astropy.nddata.UnknownUncertainty`.
+Building upon this pure container `~astropy.nddata.NDDataRef` implements:
 
-Note that a `~astropy.nddata.NDData` object is not sliceable::
++ a ``read`` and ``write`` method to access astropys unified file io interface.
++ simple arithmetics like addition, subtraction, division and multiplication.
++ slicing.
 
-    >>> ndd2[1:3]        # doctest: +SKIP
-    Traceback (most recent call last):
-        ...
-    TypeError: 'NDData' object has no attribute '__getitem__'
+Instances are created in the same way::
 
+    >>> from astropy.nddata import NDDataRef
+    >>> ndd = NDDataRef(ndd)
+    >>> ndd
+    NDDataRef([1, 2, 3, 4])
 
+But also support arithmetic (:ref:`nddata_arithmetic`) like addition::
 
-Mixins for additional functionality
------------------------------------
+    >>> import astropy.units as u
+    >>> ndd2 = ndd.add([4, 3, 2, 1] * u.erg / u.s)
+    >>> ndd2
+    NDDataRef([ 5.,  5.,  5.,  5.])
 
-Several classes are provided to add functionality to the basic ``NDData``
-container. They include:
+Because these operations have a wide range of options these are not available
+using arithmetic operators like ``+``.
 
-+ `~astropy.nddata.NDSlicingMixin` to handle slicing of N-dimensional data.
-+ `~astropy.nddata.NDArithmeticMixin` to allow arithmetic operations on
-  `~astropy.nddata.NDData` objects that include support propagation of
-  uncertainties (in limited cases).
-+ `~astropy.nddata.NDIOMixin` to use existing astropy functionality for input
-  (with the method ``read``) and output (with the method ``write``).
+Slicing or indexing (:ref:`nddata_slicing`) is possible (issuing warnings if
```

Comment:
Reminder, the sentence in the paranthesis is wrong. That would require #4271 which will not be part of 1.2 (?)